File: w06/d02/Paris/homework/mta.js
// ## Part 2 - MTA 


var lines = {n: ["ts", "34th", "28thn", "23rd", "US", "8th"], l: ["8th", "6th", "US", "3rd", "1st"], 6: ["GC", "33rd", "28th", "23rd", "US", "AP"] };

function Line(name) {
  this.name = name;
  this.lines = [];
}

var n = new Line("n", ["ts", "34th", "28thn", "23rd", "US", "8th"]);
var l = new Line("l", ["8th", "6th", "US", "3rd", "1st"]);
var six = new Line("six", ["GC", "33rd", "28th", "23rd", "US", "AP"]);

function SubwaySystem(name) {
  this.name = name;
  this.stations = [];
}

var mta = new SubwaySystem("mta", ["n", "l", "six"]);

//  prompting the user for their on/off stops and lines:

SubwaySystem.prototype.takeTrip = function() {
  var onTrain = prompt("From which line would you like to onboard?");
  var onStop = prompt("From which station would you like to onboard?");
  var offTrain = prompt("From which line would you like to get off?");
  var offStop = prompt("From which station would you like to get off?");
  Alert("Your trip will have " + totalTripLength + " stops.");
}

// Calculating the intersection point:

function getTransferPoint(onTrain, offTrain) {
  var intersection = [];
  for(var i = 0; i < onTrain.length; i++){
      for(var k = 0; k < offTrain.length; k++){
          if(onTrain[i] == offTrain[k]){
              intersection.push( onTrain[i]);
              break;
          }
      }
  }
  return intersection;
}

// calculating the distance with an if/else statement

Line.prototype.distance = function(onStop, offStop) {
  if (onTrain != offTrain) {
    //  Find out what the transfer point is
    getTransferPoint(onTrain, offTrain)
  

    // Find out the distance from our on stop to the transfer point
      var onStopIndex = mta[onTrain].indexOf(onStop);
      var onStopIntersectionIndex = mta[onTrain].indexOf(intersection);
      var beforeTransferLength = Math.abs(onStopIndex - onStopIntersectionIndex);

    // Find out the distance from our transfer point to the off stop
      var offStopIndex = mta[offTrain].indexOf(offStop);
      var offStopIntersectionIndex = mta[offTrain].indexOf(intersection);
      var afterTransferLength = Math.abs(offStopIndex - offStopIntersectionIndex);

    // Add our before transfer and after transfer stops to get the total length
      var totalTripLength = beforeTransferLength + afterTransferLength; 
  } else {
      onStopIndex = mta[onTrain].indexOf(onStop);
      offStopIndex = mta[offTrain].indexOf(offStop);
      totalTripLength = Math.abs(onStopIndex - offStopIndex);
  }
}

File: w01/d03/Marco/mta.rb
n = ['ts', '34th', '28th-n', '23rd-n', 'us', '8th']
l = ['8th-a', '6th', 'us', '3rd', '1st']
s = ['gc', '33rd', '28th-s', '23rd-s', 'us', 'a']
mta = {}
mta[:n] = n
mta[:l] = l
mta[:s] = s

################################

puts "What MTA Subway line would you like to ride to your destination? (N / L / S)" 
subway_line = gets.chomp.downcase.to_s

case subway_line

  when "n" 
    puts "Stations on the N Line: #{n.join(", ")}"

  when "l"
    puts "Stations on the L Line: #{l.join(", ")}"

  when "s"
    puts "Stations on the S Line: #{s.join(", ")}"

end

puts "Please choose a station to begin your trip."
line_start = mta[subway_line.to_sym].index(gets.chomp.downcase)

puts "Is your destination on the same Subway line? (Yes / No)"
line_switch = gets.chomp.downcase

if line_switch == "no"
  puts "What line would you like to switch to? (N, L, S)"
  line_switch_variable = gets.chomp.downcase

  puts "N Line: #{n.join(", ")}"
  puts "L Line: #{l.join(", ")}"
  puts "Six Line: #{s.join(", ")}"
  puts "Please choose a station on your destination line from the Subway stations above"

  station_switch = gets.chomp.downcase
  line_end_diff = mta[subway_line.to_sym].index(line_switch_variable)

  puts "You have #{(line_start - mta[:"#{subway_line}"].index(line_start)).abs + (mta[:"#{subway_line}"].index(line_switch_variable) - line_end_diff).abs} stops on your trip"

#intersection = ""

else
  puts "Which stop is your destination?"
  line_end_same = mta[:"#{subway_line}"].index(gets.chomp.downcase)
  puts "You have #{(line_end_same - line_start).abs} stops on your trip"
end